src/handlers/canonicalFieldsHandlers.ts

```diff
@@ -33,6 +45,75 @@ const getCanonicalFields = (req: Request, res: Response): void => {
     });
 };
 
+const postCanonicalFieldBodySchema: JSONSchemaType<Omit<CanonicalField, 'createdAt' | 'id'>> = {
+  type: 'object',
+  properties: {
+    label: {
+      type: 'string',
+    },
+    shortCode: {
+      type: 'string',
+    },
+    dataType: {
+      type: 'string',
+    },
+  },
+  required: [
+    'label',
+    'shortCode',
+    'dataType',
+  ],
+};
+const isPostCanonicalFieldBody = ajv.compile(postCanonicalFieldBodySchema);
```

Comment:
I haven't tested it but will this compilation take place at server startup or at first POST to /canonicalFields? If the latter, for more consistent performance we might want to go to the common area. If it only takes a few milliseconds regardless then this is a non-issue.